File: lista2-thehuxley/brincandodeleet.py
def codificar(string, codigos):
    for i in range(len(string)):
        for codigo in codigos:
            if(string[i] in codigo):
                string = string.replace(string[i], codigo[2])
    string = inverter(string)
    return string

def contarSubstituicoes(string, codigos):
    substituicoes = 0
    for i in range(len(string)):
        for codigo in codigos:
            if(string[i] in codigo):
                substituicoes += 1
    return substituicoes

def stringValida(string):
    if(isEmpty(string)):
        print("vazio")
        print("0")
    elif(hasNumber(string)):
        print("numeros")
        print("0")
    else:
        return True

def isEmpty(string):
    if(string == ''):
        return True

def hasNumber(string):
    numeros = ['0', '1', '2', '3', '4', '5', '6', '7', '8', '9']
    for caracter in string:
        if(caracter in numeros):
            return True

def inverter(txt):
  return txt[::-1]

a = ['a','A', '@']
e = ['e', 'E', '3']
i = ['i', 'I', '1']
o = ['o', 'O', '0']
t = ['t', 'T', '7']
s = ['s', 'S', '5']
codigos = [a, e, i , o, t, s]

string = input()
if(stringValida(string)):
    leet = codificar(string, codigos)
    substituicoes = contarSubstituicoes(string, codigos)
    print(leet)
    print(substituicoes)

File: Lua/brincandodeleet.lua
function isEmpty(string)
    if(string == "") then
        return true
    else
        return false
    end
end

function hasNumber(string)
    if(string.match(string, "%d")) then -- ola12 mundo
        return true
    else
        return false
    end
end

function codificar(string, codigo)
    substituicoes = 0
    
    for key, value in pairs(codigo) do
        string, sub = string.gsub(string, key, value)
        substituicoes = substituicoes + sub
    end

    return string.reverse(string), substituicoes    
end

codigo = {
    a = "@",
    A = "@",
    e = "3",
    E = "3",
    i = "1",
    I = "1",
    o = "0",
    O = "0",
    t = "7",
    T = "7",
    s = "5",
    S = "5"
}

string = io.read()
substituicoes = 0

if(isEmpty(string)) then
    print("Vazio.")
elseif(hasNumber(string)) then
    print("Numeros.")
else
    leet, substituicoes = codificar(string, codigo)
    print(leet)
end
    print(substituicoes)